### PR TITLE
Added partial fix for #161

### PR DESCRIPTION
**Pull request by joesan**
URL's that does not contain a scheme, but contains a @ are being treated as E-Mail. So added a Regex to check for E-Mail

### CODE REVIEW

**Comment by mre** [comment]
Thanks for the PR and for helping out with #161 @joesan. Much appreciated!
To be honest I'm not totally sold on the regex implementation yet. E-Mails are notoriously hard to parse and me might miss an edge case. There is a great talk and a nice blog post about the topic [here](https://github.com/kdeldycke/awesome-falsehood#emails).
If we see this as a temporary workaround that will be removed once https://github.com/robinst/linkify/pull/8 is merged, it would be fine from my side. In that case we should add a comment to the pull request to the code so we won't forget. 😉
What's your opinion?

**Comment by joesan** [comment]
> Thanks for the PR and for helping out with #161 @joesan. Much appreciated!
> To be honest I'm not totally sold on the regex implementation yet. E-Mails are notoriously hard to parse and me might miss an edge case. There is a great talk and a nice blog post about the topic [here](https://github.com/kdeldycke/awesome-falsehood#emails).
> If we see this as a temporary workaround that will be removed once [robinst/linkify#8](https://github.com/robinst/linkify/pull/8) is merged, it would be fine from my side. In that case we should add a comment to the pull request to the code so we won't forget. 
> What's your opinion?

Valid point. E-Mail addresses are meant to be created such that they are simple enough to be addressed. I see that the RFC allows for an E-Mail like this:

`phil.h\@\@ck@haacked.com`

But I would not want to send any E-Mail to such an address that one looks like a hacking endpoint. But nevertheless, you are right. In this case, I would not merge this PR as I do not like adding something with comments saying that this will be removed after something else is fixed. So going by the strict E-Mail validation using the RFC, I would say let us skip this PR.

**Comment by mre** [comment]
Okay then, let's close this. The unit tests might be valuable in the future, so we should revisit this in the future.
